File: benchmarks/cpp/file operations/file_write.cpp
#include <fstream>
#include <chrono>
#include <iostream>

using namespace std;
using namespace chrono;

int main() {
    const long long SIZE = 100 * 1024 * 1024; // change size here
    ofstream file("write_test.bin", ios::binary);

    char buffer[4096] = {0};
    long long written = 0;

    auto start = high_resolution_clock::now();

    while (written < SIZE) {
        file.write(buffer, sizeof(buffer));
        written += sizeof(buffer);
    }

    file.flush(); // ensure data is written

    auto end = high_resolution_clock::now();
    duration<double> elapsed = end - start;

    cout << "C++ File Write Time: " << elapsed.count() << " seconds\n";
    file.close();
    return 0;
}

File: benchmarks/rust/file operations/file_write.rs
use std::fs::File;
use std::io::Write;
use std::time::Instant;

fn main() {
    let size: u64 = 100 * 1024 * 1024; // change size here
    let mut file = File::create("write_test.bin").unwrap();

    let buffer = [0u8; 4096];
    let mut written: u64 = 0;

    let start = Instant::now();

    while written < size {
        file.write_all(&buffer).unwrap();
        written += buffer.len() as u64;
    }

    file.flush().unwrap();

    let elapsed = start.elapsed();
    println!("Rust File Write Time: {:.6} seconds", elapsed.as_secs_f64());
}
